src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -4284,7 +4284,10 @@ static ref int M(ref int i)
             CreateCompilationWithMscorlib46(text).VerifyDiagnostics(
                 // (6,9): error CS8150: By-value returns may only be used in methods that return by value
                 //         return;
-                Diagnostic(ErrorCode.ERR_MustHaveRefReturn, "return").WithLocation(6, 9)
+                Diagnostic(ErrorCode.ERR_MustHaveRefReturn, "return").WithLocation(6, 9),
+                // (6,9): error CS0126: An object of a type convertible to 'int' is required
+                //         return;
+                Diagnostic(ErrorCode.ERR_RetObjectRequired, "return").WithArguments("int").WithLocation(6, 9)
```

Comment:
It feels like most diagnostics that are added by this change are "fine", except, when the return statement's ref kind doesn't match the containing method signature, the extra diagnostic for conversion failure is not "ideal". But, I think that additional investment to avoid reporting this diagnostic isn't necessary.

It *is* necessary, IMO, to create the conversion anyway, so that IDE can get complete information. Even if the return statement has a wrong ref kind, we should still indicate the converted type through the public API.